### PR TITLE
http-client-java, bump typespec libs

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
bump for the new typespec versions

1. bug fix for pagination (when no `nextLink`)
2. improvement for unbranded error without body
3. remove `core-json` from unbranded, as it recently merged to `core`

### CODE REVIEW

File: packages/http-client-java/emitter/src/code-model-builder.ts [hunk 1]
```diff
@@ -969,20 +969,19 @@ export class CodeModelBuilder {
         if (bodyType && bodyType.kind === "model") {
           const itemName = sdkMethod.response.resultPath;
           const nextLinkName = sdkMethod.nextLinkPath;
-          if (itemName && nextLinkName) {
-            op.extensions = op.extensions ?? {};
-            op.extensions["x-ms-pageable"] = {
-              itemName: itemName,
-              nextLinkName: nextLinkName,
-            };
```
Comment (weidongxu-microsoft) [inline packages/http-client-java/emitter/src/code-model-builder.ts:977]: Fix bug introduced on tcgc migration.

itemName and nextLinkName is not required for pageable op.

File: packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/template/ProxyTemplate.java [hunk 2]
```diff
@@ -211,9 +211,15 @@ protected void writeUnexpectedExceptions(ProxyMethod restAPIMethod, JavaInterfac
...
-                        + " }, exceptionBodyClass = " + errorModel.getName() + ".class)");
+                    if (errorModel == null) {
+                        interfaceBlock.annotation("UnexpectedResponseExceptionDetail(statusCode = {"
+                            + exception.getValue().stream().map(String::valueOf).collect(Collectors.joining(","))
+                            + " })");
+                    } else {
+                        interfaceBlock.annotation("UnexpectedResponseExceptionDetail(statusCode = {"
+                            + exception.getValue().stream().map(String::valueOf).collect(Collectors.joining(","))
+                            + " }, exceptionBodyClass = " + errorModel.getName() + ".class)");
+                    }
```
Comment (weidongxu-microsoft) [inline packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/template/ProxyTemplate.java:222]: handle `@error` without body, in unbranded.

File: packages/http-client-java/generator/http-client-generator-test/src/main/java/tsptest/armresourceprovider/fluent/models/OperationInner.java [hunk 3]
```diff
@@ -45,7 +45,7 @@ public final class OperationInner {
     /*
      * Extensible enum. Indicates the action type. "Internal" refers to actions that are for internal only APIs.
      */
-    @JsonProperty(value = "actionType")
+    @JsonProperty(value = "actionType", access = JsonProperty.Access.WRITE_ONLY)
```
Comment (weidongxu-microsoft) [inline packages/http-client-java/generator/http-client-generator-test/src/main/java/tsptest/armresourceprovider/fluent/models/OperationInner.java:48]: Code change via regen.

Appears matches latest Azure.ResourceManager https://github.com/Azure/typespec-azure/blob/main/packages/typespec-azure-resource-manager/lib/common-types/types.tsp#L145-L158

File: packages/http-client-java/generator/http-client-generator-test/tsp/visibility.tsp [hunk 4]
```diff
@@ -10,8 +10,8 @@ using Azure.ClientGenerator.Core;
 namespace TspTest.Visibility;
 
 model Dog {
-  @visibility("read") id: int32;
-  @visibility("update") secretName: string;
+  @visibility(Lifecycle.Read) id: int32;
```
[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply packages/http-client-java/generator/http-client-generator-test/tsp/visibility.tsp:13]: I've updates the whole test case.

Previously we do a 
```
@withVisibility(Lifecycle.Read)
model ReadDog {
  ...Dog;
}
```

And then use it as an input. It seems not reasonable.

File: packages/http-client-java/generator/http-client-generator-test/tsp/response.tsp [hunk 5]
```diff
@@ -86,26 +86,20 @@ model StringsList {
...
   nextLink?: string;
 }
 
 @pagedResult
 model Int32sList {
   @items
   value: int32[];
-
-  @global.Azure.Core.nextLink
-  nextLink?: string;
```
Comment (weidongxu-microsoft) [inline packages/http-client-java/generator/http-client-generator-test/tsp/response.tsp:99]: Updated to test pagination without `nextLink`.

**Review by XiaofeiCao — APPROVED** [review]

**Pull request merged**